vue/sbc-common-components/src/services/token.services.ts

```diff
@@ -69,20 +74,26 @@ class TokenServices {
 
   private scheduleRefreshToken (refreshEarlyTimeinMilliseconds:number) {
     let self = this
-
+    let refreshTokenExpiresIn = -1
     // check if refresh token is still valid . Or else clear all timers and throw errors
-    let refreshTokenExpiresIn = self.kc.refreshTokenParsed['exp'] - Math.ceil(new Date().getTime() / 1000) + self.kc.timeSkew
+    if (self.kc !== undefined && self.kc.timeSkew !== undefined && self.kc.refreshTokenParsed !== undefined) {
+      refreshTokenExpiresIn = self.kc.refreshTokenParsed['exp']! - Math.ceil(new Date().getTime() / 1000) + self.kc.timeSkew
+    }
     if (refreshTokenExpiresIn < 0) {
       throw new Error('Refresh Token Expired..No more token refreshes')
     }
-
-    let expiresIn = self.kc.tokenParsed['exp'] - Math.ceil(new Date().getTime() / 1000) + self.kc.timeSkew
+    let expiresIn = -1
+    if (self.kc !== undefined && self.kc.tokenParsed !== undefined && self.kc.tokenParsed['exp'] !== undefined) {
+      expiresIn = self.kc.tokenParsed['exp'] - Math.ceil(new Date().getTime() / 1000) + self.kc!.timeSkew!
```

Comment:
Do you need `self.kc!` non-null assertion operator here?  Your condition on line 86 should be sufficient.

Reply:
done